hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceFactory.java

```diff
@@ -29,4 +29,6 @@ HazelcastInstance newHazelcastInstance(Config config, String instanceName,
                                            NodeContext nodeContext) throws Exception;
 
     HazelcastInstance newHazelcastInstance(Config config) throws Exception;
+
+    HazelcastInstanceProxy newHazelcastInstanceProxy(HazelcastInstanceImpl hazelcastInstance);
```

Comment:
protected HazelcastInstanceProxy(HazelcastInstanceImpl original)
Proxy constructor asks me to pass here impl.
protected HazelcastInstanceProxy(HazelcastInstanceImpl original)

I have the same for JET

protected JetHazelcastInstanceProxy(HazelcastInstanceImpl original)